File: CS/GridControlWithResizingColumnHeader/Custom GridView/MyGridView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;

namespace GridControlWithFindPanel
{
    public class MyGridView : GridView
    {
        public MyGridView()
            : this(null)
        {
        }

        public MyGridView(GridControl grid)
            : base(grid)
        {
            // put your initialization code here
        }

        protected override string ViewName
        {
            get
            {
                return "MyGridView";
            }
        }

        protected override void EndRowSizing()
        {
            int rowHandle = (int)(Painter.ReSizingObject);
            if (rowHandle > 0)
            {
                GridRowInfo ri = ViewInfo.GetGridRowInfo(rowHandle);
                if (ri == null) return;
                int height = (Painter.CurrentSizerPos - ri.DataBounds.Top) / ViewInfo.RowLineCount;
                if (height > 0)
                    RowHeight = height;
            }
            else
            {
                GridColumnsInfo ci = ViewInfo.ColumnsInfo;
                if (ci == null) return;
                int height = (Painter.CurrentSizerPos - ci.FirstColumnInfo.Bounds.Top) / ViewInfo.RowLineCount;
                if (height > 0)
                    ColumnPanelRowHeight = height;
            }
        }
    }
}

File: VB/GridControlWithResizingColumnHeader/Custom GridView/MyGridView.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo

Namespace GridControlWithFindPanel
    Public Class MyGridView
        Inherits GridView

        Public Sub New()
            Me.New(Nothing)
        End Sub

        Public Sub New(ByVal grid As GridControl)
            MyBase.New(grid)
            ' put your initialization code here
        End Sub

        Protected Overrides ReadOnly Property ViewName() As String
            Get
                Return "MyGridView"
            End Get
        End Property

        Protected Overrides Sub EndRowSizing()
            Dim rowHandle As Integer = CInt((Painter.ReSizingObject))
            If rowHandle > 0 Then
                Dim ri As GridRowInfo = ViewInfo.GetGridRowInfo(rowHandle)
                If ri Is Nothing Then
                    Return
                End If
                Dim height As Integer = (Painter.CurrentSizerPos - ri.DataBounds.Top) / ViewInfo.RowLineCount
                If height > 0 Then
                    RowHeight = height
                End If
            Else
                Dim ci As GridColumnsInfo = ViewInfo.ColumnsInfo
                If ci Is Nothing Then
                    Return
                End If
                Dim height As Integer = (Painter.CurrentSizerPos - ci.FirstColumnInfo.Bounds.Top) / ViewInfo.RowLineCount
                If height > 0 Then
                    ColumnPanelRowHeight = height
                End If
            End If
        End Sub
    End Class
End Namespace